### PR TITLE
Media group role refacto

### PR DESCRIPTION
**Pull request by Thiblef**
[OO-FEATURE] Media roles gestion for each folders are available in the group panel
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1416
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/163
https://github.com/open-orchestra/open-orchestra-model-interface/pull/163
https://github.com/open-orchestra/open-orchestra/pull/831

### CODE REVIEW

**Comment by Thiblef** [comment]
@itkg-alavieille @itkg-ngilain @itkg-nanne @nicolasThal  PR updated

File: MediaAdmin/Security/Authorization/Voter/MediaFolderGroupRoleVoter.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    /**
+     * Returns the vote for the given parameters.
+     *
+     * This method must return one of the following constants:
+     * ACCESS_GRANTED, ACCESS_DENIED, or ACCESS_ABSTAIN.
+     *
+     * @param TokenInterface $token A TokenInterface instance
+     * @param FolderInterface|null $object The object to secure
```
Comment (alavieille) [inline MediaAdmin/Security/Authorization/Voter/MediaFolderGroupRoleVoter.php]: align php doc

File: MediaAdminBundle/EventSubscriber/MediaFolderGroupTransformSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * @param GroupFacadeEvent $event
+     */
+    public function postGroupReverseTransformation(GroupFacadeEvent $event)
+    {
+        $facade = $event->getGroupFacade();
+        $group = $event->getGroup();
+
+        foreach ($facade->getDocumentRoles() as $documentRoleFacade) {
+            if ('folder' === $documentRoleFacade->type) {
```
Comment (alavieille) [inline MediaAdminBundle/EventSubscriber/MediaFolderGroupTransformSubscriber.php]: add it in the constant

**Comment by Thiblef** [comment]
@itkg-alavieille @itkg-ngilain @itkg-nanne @nicolasThal PR updated

File: MediaAdminBundle/EventSubscriber/MediaFolderGroupTransformSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * @param GroupFacadeEvent $event
+     */
+    public function postGroupReverseTransformation(GroupFacadeEvent $event)
+    {
+        $facade = $event->getGroupFacade();
+        $group = $event->getGroup();
+        if (!$this->transformer instanceof TransformerWithGroupInterface) {
+            throw new UnexpectedValueException("Document Group Role Transformer must be an instance of TransformerWithGroupInterface");
```
Comment (alavieille) [inline MediaAdminBundle/EventSubscriber/MediaFolderGroupTransformSubscriber.php]: Media Group Role Transformer must be ...

File: MediaAdminBundle/Transformer/FolderTransformer.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        $facade = $this->newFacade();
+
+        $facade->folderId = $folder->getId();
+        $facade->name = $folder->getName();
+        $facade->createdAt = $folder->getCreatedAt();
+        $facade->updatedAt = $folder->getUpdatedAt();
+        if ($folder->getParent() instanceof FolderInterface) {
+            $facade->parentId = $folder->getParent()->getId();
+        } else {
+            $facade->parentId = "-";
```
Comment (alavieille) [inline MediaAdminBundle/Transformer/FolderTransformer.php]: i don't understand this line, when a folder is in root her parentId is `-`

File: MediaAdminBundle/Resources/views/Tree/menu_macros_bo.html.twig [hunk 5]
```diff
@@ -9,7 +9,6 @@
 {% endmacro %}
 
 {% macro new_folder(parentId) %}
-    {% if is_granted('ROLE_ACCESS_CREATE_MEDIA_FOLDER') %}
         <li>
```
Comment (ngilain) [inline MediaAdminBundle/Resources/views/Tree/menu_macros_bo.html.twig]: You should re-ident this block

File: MediaAdminBundle/Transformer/FolderTransformer.php (hunk 4 above)
Comment (ngilain) [inline MediaAdminBundle/Transformer/FolderTransformer.php]: Maybe you could set it to null or use a class constant instead

Comment (Thiblef) [inline MediaAdminBundle/Transformer/FolderTransformer.php]: I chose to use the same notation than nodes to have generic views ( like the `groupTreeForm._tpl.twig`). I'll set it into a constant.

**Comment by Thiblef** [comment]
@itkg-alavieille @itkg-ngilain

File: MediaAdminBundle/EventSubscriber/MediaFolderGroupTransformSubscriber.php [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+/**
+ * Class MediaFolderGroupTransformSubscriber
+ */
+class MediaFolderGroupTransformSubscriber implements EventSubscriberInterface
+{
+    protected $router;
+    protected $transformer;
+
+    /**
+     * @param UrlGeneratorInterface $router
```
Comment (alavieille) [inline MediaAdminBundle/EventSubscriber/MediaFolderGroupTransformSubscriber.php]: update php doc

File: MediaAdminBundle/Transformer/FolderTransformer.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;
+use OpenOrchestra\Media\Model\FolderInterface;
+use OpenOrchestra\MediaAdminBundle\Facade\FolderFacade;
+
+/**
+ * Class FolderTransformer
+ */
+class FolderTransformer extends AbstractTransformer
+{
+    const ROOT_FOLDERS_PARENT_ID = '-';
```
Comment (alavieille) [inline MediaAdminBundle/Transformer/FolderTransformer.php]: i think is better to move this constant in FolderInterface and use it also in template twig

Comment (Thiblef) [inline MediaAdminBundle/Transformer/FolderTransformer.php]: The twig template is part of cms-bundle, I can't use a media-admin-bundle constant there.

**Comment by Thiblef** [comment]
@itkg-alavieille @itkg-ngilain @itkg-nanne @nicolasThal PR updated

**Comment by alavieille** [comment]
:+1:  when test run

**Comment by Thiblef** [comment]
@itkg-alavieille @itkg-ngilain PR updated, I had forgotten to add a message in group media panel when group site was'nt defined. You can validate only the last commit ;)

**Comment by alavieille** [comment]
:+1:

**Pull request merged**